instrumentation/servlet/servlet-common/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/servlet/ExperimentalSnippetHolder.java

```diff
@@ -9,8 +9,14 @@ public class ExperimentalSnippetHolder {
 
   private static volatile String snippet = "";
 
+  private static boolean isSet = false;
+
   public static void setSnippet(String snippet) {
+    if (isSet) {
+      return;
+    }
```

Comment:
Consider changing the type of `snippet` to `AtomicReference<String>` and using `compareAndSet` instead